yt_dlp/extractor/facebook.py

```diff
@@ -493,14 +509,19 @@ def process_formats(info):
 
         def extract_relay_data(_filter):
             return self._parse_json(self._search_regex(
-                r'handleWithCustomApplyEach\([^,]+,\s*({.*?%s.*?})\);' % _filter,
+                r'data-sjs>({.*?%s.*?})<\/script' % _filter,
                 webpage, 'replay data', default='{}'), video_id, fatal=False) or {}
 
         def extract_relay_prefetched_data(_filter):
             replay_data = extract_relay_data(_filter)
             for require in (replay_data.get('require') or []):
                 if require[0] == 'RelayPrefetchedStreamCache':
                     return try_get(require, lambda x: x[3][1]['__bbox']['result']['data'], dict) or {}
+                else:
+                    data = traverse_obj(require, (
+                        ..., ..., '__bbox', 'require', lambda _, v: "RelayPrefetchedStreamCache" in v,
+                        ..., ..., "__bbox", "result", "data"))
+                    return data[0] if data else {}
```

Comment:
```suggestion
                else:
                    return traverse_obj(require, (
                        ..., ..., '__bbox', 'require', lambda _, v: 'RelayPrefetchedStreamCache' in v,
                        ..., ..., '__bbox', 'result', 'data', {dict}), get_all=False) or {}
```
Do we need the `if/else` though? From testing a bit, it seems like the `if require[0]...` branch might be obsolete now. I could be wrong

Reply:
facebook has so many edge cases. It's best to not remove old patterns

Reply:
It LOOKS obsolete indeed. Is there any common function that would allow finding:
```
lambda _, v: 'RelayPrefetchedStreamCache' in v, ..., ..., '__bbox', 'result', 'data', {dict}
```
as deep as needed to cover both of these cases in 1?

Reply:
> Is there any common function that would allow finding...

https://github.com/yt-dlp/yt-dlp/pull/7890/files#r1299998111